### ISSUE TITLE
`iss` validation should be against a collection ("whitelist")

### ISSUE THREAD

**Issue by delitescere** [funcool/buddy-sign#47]
From the perspective of the caller of validating the incoming JWT, they allow themselves (a single audience) of possibly many audiences listed in the claim.

Conversely, they also may allow tokens generated a number of issuers, one of which is presented in the claim.

So, the validation logic should accept a single audience (to match against a list of audiences in the token, as it does now), and also one or a list of issuers (to match against the single issuer in the token).

Entails changing https://github.com/funcool/buddy-sign/blob/master/src/buddy/sign/jwt.clj#L26 to something that is similar to the `aud` validation, but with inverse item-to-list matching.

**Comment by delitescere** [comment]
Something like this should do it:
```
  (when (and iss-allowed (let [iss-claim (:iss claims)]
                          (if (coll? iss-allowed)
                            (not-any? #{iss-claim} iss-allowed)
                            (not= iss-claim iss-allowed))))
    (throw (ex-info (str "Issuer does not match " iss-allowed)
                    {:type :validation :cause :iss}))
```

**Comment by niwinz** [comment]
Fixed by #48
